### ISSUE TITLE
Allow keyboard scrolling of `ScrollContainer`s by default

### ISSUE THREAD

**Issue by romlok** [godotengine/godot-proposals#7365]
### Describe the project you are working on

A UI to show a scrollable list of credits

### Describe the problem or limitation you are having in your project

A focused `ScrollContainer` doesn't respond to any keyboard input.
One needs to use a mouse or touch gesture to scroll UI areas which don't otherwise contain any focusable controls (with which one can use `follow_focus`).

I found several issues along this path:

1. `ScrollContainer` isn't focusable by default (though this is arguable either way imo)
2. `ScrollContainer` has no focus indicator - in the default theme, at least
3. When `ScrollContainer` does have focus, it does not respond to keyboard input
4. `ScrollBar`s, either embedded in a `ScrollContainer` or not, *do* respond to keyboard input, but are not focusable by default
5. A `ScrollContainer`'s embedded scrollbars can only have their `focus_mode` changed via code
6. An embedded scrollbar has its focus ring clipped by the `ScrollContainer` ![Screenshot_20230724_162048](https://github.com/godotengine/godot-proposals/assets/441188/be343fbe-62f2-40c9-b4e5-4b543b84d9de)
7. The default `step` for a scrollbar is 0 - which disables up and down arrows out of the box
8. `ScrollBar`s don't respond to page up and page down (despite having a `page` property! :see_no_evil:)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The `ScrollContainer` control should by default respond to keyboard input for scrolling in the same manner as a stand-alone `[H|V]ScrollBar`.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Minimally, `[vertical|horizontal]_scroll_focus_mode` properties could be added to `ScrollContainer`, which would proxy the `focus_mode` on the respective embedded `ScrollBar`s.
With no other changes, I believe the embedded scrollbars would look a bit bad when focused, but they'd work.

Alternatively, it may be more seamless for `ScrollContainer`s themselves to relay input events to their embedded scrollbars. That way the entire scrollable area could be focus-highlighted, rather than just the individual scrollbar/s.
Though that would raise the question of how home/end and page up/down work when both scrollbars are present :confused: 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It _can_ be worked around with scripting - by manually (re)implementing scrolling, or explicitly setting the embedded `ScrollBar`s' `focus_mode`; and it _may_ not be used often.

However, when it _is_ needed, it can be make-or-break for a UI's keyboard accessibility; and such functionality is easy to overlook for a developer who primarily uses a mouse or touch interface when testing their app.  So having this be out-of-the-box behaviour would be beneficial.

### Is there a reason why this should be core and not an add-on in the asset library?

This concerns the behaviour of a built-in node.

**Comment by felaugmar** [comment]
I agree this could become a nice enhancement, but not as the default behavior. 

See the already implemented `follow_focus`, it's useful many times, but it's disabled by default and it works fine like that.

**Comment by Psychpsyo** [comment]
> Alternatively, it may be more seamless for `ScrollContainer`s themselves to relay input events to their embedded scrollbars. That way the entire scrollable area could be focus-highlighted, rather than just the individual scrollbar/s. Though that would raise the question of how home/end and page up/down work when both scrollbars are present 😕

I just checked how web browsers handle this case:
Chrome and Firefox agree that keyboard scrolls (Home/End/PgUp/PgDown) always go up and down, even if there is a horizontal scrollbar (if there is *only* a horizontal scrollbar, keyboard scrolling does nothing). Holding shift does not make them scroll horizontally, it instead extends text selections to the start/end of the line with Home/End. This is despite the fact that shift+mouse wheel does allow scrolling left & right.
Safari does basically the same, except that it keyboard (but not mouse) scrolls horizontally instead of vertically when the container's writing mode is vertical (= text flowing top to bottom, with lines being stacked next to one another). And also that it doesn't do the thing with text selection.

TL;DR: I think making a 2-axis scroll container always keyboard-scroll up and down should not be that problematic. But I personally also quite like the idea of using shift plus those buttons to make it scroll sideways. (Although I'm not sure if there is any precedent for that anywhere.)
